lib/IMAP/FolderMapper.php

```diff
@@ -107,6 +107,13 @@ public function buildFolderHierarchy(array $folders, bool $havePrefix): array {
 			}
 		}
 
+		//filter for \\noselect
+        foreach ($top as $folder) {
```

Comment:
please use `array_filter` instead, removing elements of an array you're currently iterating on is scary :see_no_evil:

Reply:
Indentation is a bit off, btw

Reply:
What happens if I have a top-level mailbox that is not selectable with subfolders that are? Will the subfolders still be shown?